stream/visualization/schedule.py

```diff
@@ -38,17 +39,17 @@
 
 
 def plot_timeline_brokenaxes(
-    scme,  # StreamCostModelEvaluation
-    draw_dependencies: object = True,
-    section_start_percent: object = (0, 50, 95),
-    percent_shown: object = (5, 5, 5),
-    plot_data_transfer: object = False,
-    fig_path: object = "outputs/schedule_plot.png",
-) -> object:
-    G: DiGraph = scme.workload
+    scme: "StreamCostModelEvaluation",
+    draw_dependencies: bool = True,
+    section_start_percent: tuple[int, ...] = (0, 50, 95),
+    percent_shown: tuple[int, ...] = (5, 5, 5),
+    plot_data_transfer: bool = False,
+    fig_path: str = "outputs/schedule_plot.png",
+) -> None:
+    G: Workload = scme.workload
     accelerator: Accelerator = scme.accelerator
 
-    nb_layers = len(set(iter([n.id[0] for n in G.nodes()])))
+    nb_layers = len(set(iter([n.id for n in G.nodes()])))
```

Comment:
Does this not change the logical meaning of the code?

Stream uses a tuple id of the form `(layer, chunk)`. In the case of layer fusion, `layer` is ... the layer... and `chunk` is the sub-layer split.

The code previously only checked for the number of layers but now it effectively checks for the number of computational nodes

Reply:
Ah nvm, I notice now that there is a `sub_id` field